### PR TITLE
bug fixed

### LINKED ISSUE #2478: there may be some problems with  _.debounce
in this file <https://github.com/jashkenas/underscore/blob/master/underscore.js>

I think there may be some issues in _.debounce function, if we use like this

```
var lazyLayout = _.debounce(function() {
	console.log(123)
	} , 300, true);

window.onresize = lazyLayout;
```

there will be only one '123' in the console after resizing the window several times.

### PR DESCRIPTION
**Pull request by hanzichi**
#2478

### CODE REVIEW

File: underscore.js [hunk 1]
```diff
@@ -865,7 +865,7 @@
       if (callNow) {
         timeout = setTimeout(later, wait);
         result = func.apply(this, args);
-      } else if (!immediate) {
```
Comment (jridgewell) [inline underscore.js]: This will schedule a new timeout every time the function is called when `immediate = true`.

**Comment by jridgewell** [comment]
After review, this is definitely incorrect behavior. This will call `func` after `wait` every time the debounced function is called when `immediate = true`.

**Comment by hanzichi** [comment]
@jridgewell  thanks for your reply!  I just wonder with the code below, I just find the different results when resizing the window for several times using [underscore.js](https://github.com/jashkenas/underscore/blob/master/underscore.js) and [underscore-min.js](https://github.com/jashkenas/underscore/blob/master/underscore-min.js)

```
  var lazyLayout = _.debounce(function() {
    	console.log(123)
    	} , 300, true);

    window.onresize = lazyLayout;
```




don't you? hope to hear from you soon ! thanks !

**Comment by megawac** [comment]
@hanzichi are you mixing up `debounce` with `throttle`?

**Comment by hanzichi** [comment]
@megawac no, i am not, could you pls have a try as i said above ?

**Comment by megawac** [comment]
I did and I got the results I expected

**Comment by hanzichi** [comment]
i went through the code, and found line 866

`    timeout = setTimeout(later, wait);`

but we will never trigger `later

**Comment by jridgewell** [comment]
> i went through the code, and found line 866
> timeout = setTimeout(later, wait);

That's `_.throttle`'s code.

**Comment by hanzichi** [comment]
I do not think so...

**Comment by jridgewell** [comment]
I don't think this is the right solution, but there's definitely something wrong. I'll spend some more time on this tomorrow.

**Comment by hanzichi** [comment]
@jridgewell thanks for your work !

**Comment by jridgewell** [comment]
Closing this for #2482. Thanks for the bug report!

**Comment by hanzichi** [comment]
that's what i should do, thanks for your quick fix again !

**Pull request closed without merging**